### DISCUSSION TITLE
Coqui AI TTS Russian Model Training Walkthrough

### DISCUSSION THREAD

**Discussion by iprovalo** [coqui-ai/TTS#1701] [General]
I ran a few training experiments on a Russian language ljspeech dataset ([M-AI-Labs](https://www.caito.de/2019/01/03/the-m-ailabs-speech-dataset/)) using Coqui AI TTS.  So far I have been training Glow TTS with MB Melgan and Hifigan vocoders.  I will be adding the Tacotron and Fast Speech to the list later.  While this is still a work-in-progress, here are some preliminary results which I would like to share for the comments and suggestions.

I documented the configs, as well as some key charts and metrics from the tensorboard along with some observations for nine jobs - two vocoders, and seven models.  

[Google Doc version](https://docs.google.com/document/d/1f2IC66kuogjl9Omjkbz2Ixzaw5K1iFXMNNaBjITDQw0/edit?usp=sharing), open for the comments.

My open questions:

1. Why is mailabs dataset style not working as well as ljspeech style?
2. Why is ljspeech dataset alignment not looking as smooth or diagonal as mailabs one?  Is it the padding?
3. Curious why the batch size increase causes training slowdown.  Should it be the opposite?
4. Why does the average loss in Run 9 for Hifigan look spiky?

Thank you!

**Comment by iprovalo** [comment]
Related to https://github.com/coqui-ai/TTS/discussions/1703

**Reply by iprovalo** [reply]
Added RUSLAN dataset runs for the model and vocoder.  Much faster and better quality so far.

**Reply by MaksimSerg** [reply]
Can you show samples of synthesized speech from paragraphs 12 and 13 of the "Updated Table"?

**Reply by iprovalo** [reply]
Here is a [google drive folder](https://drive.google.com/drive/folders/1bZO3a7TaFFM_8qMknWLfjGn1NdPxY0LZ) with the samples:

- Tensorboard:
tensorboard_glow_tts_run_12.wav
tensorboard_mb_melgan_run_13.wav

- Inference based on the based best model:
glow_tts_ru_ruslan.wav
glow_tts_ru_ruslan_no_vocoder.wav

**Reply by MaksimSerg** [reply]
tensorboard_mb_melgan_run_13.wav - это Glow-tts + multiband_melgan ?

**Reply by iprovalo** [reply]
No, this one is just mb_melgan, no model.

**Reply by kin0303** [reply]
@iprovalo 

Hi, I want to ask. How do you train the vocoder if you use multi-speaker? is it training using one of the speakers?

**Reply by iprovalo** [reply]
@blackmamba1122 yes, that's how I would do it (pick one speaker).  I am trying that for the Spanish dataset at the moment.  

Adding @erogol to comment.

**Comment by madey83** [comment]
@iprovalo

durring your test did you train Hifigan vocoders on russian dataset or you used what is available on github page: https://github.com/jik876/hifi-gan 


What was your final config to get the best audio? Can you share it?
